### PR TITLE
sys/vm: In the implied PROT_MAX case, filter out requests based on prot.

### PR DESCRIPTION
**Pull request by bukinr**
this is needed for firefox.
tested with firefox 107 on cheribsd hybrid

### CODE REVIEW

File: sys/vm/vm_mmap.c [hunk 1]
```diff
@@ -772,14 +776,14 @@ kern_mmap(struct thread *td, const struct mmap_req *mrp)
...
 		 */
 		cap_rights_init_one(&rights, CAP_MMAP);
-		if (max_prot & PROT_READ)
+		if (cap_prot & PROT_READ)
 			cap_rights_set_one(&rights, CAP_MMAP_R);
 		if ((flags & MAP_SHARED) != 0) {
-			if (max_prot & PROT_WRITE)
+			if (cap_prot & PROT_WRITE)
 				cap_rights_set_one(&rights, CAP_MMAP_W);
+			if (cap_prot & PROT_EXEC)
```
Comment (jrtc27) [inline sys/vm/vm_mmap.c:784]: You shouldn't need to move this any more

File: sys/vm/vm_mmap.c [hunk 2]
```diff
@@ -788,9 +792,9 @@ kern_mmap(struct thread *td, const struct mmap_req *mrp)
 			error = EINVAL;
 			goto done;
 		}
-		if ((max_prot & cap_maxprot) != max_prot) {
+		if ((cap_prot & cap_maxprot) != cap_prot) {
```
Comment (jrtc27) [inline sys/vm/vm_mmap.c:795]: I do wonder if we want to do a max_prot &= cap_maxprot after this point...

File: sys/vm/vm_mmap.c (hunk 1 above)
Comment (bukinr) [reply sys/vm/vm_mmap.c]: right, I testted it still works without the move

File: sys/vm/vm_mmap.c (hunk 2 above)
Comment (bukinr) [reply sys/vm/vm_mmap.c:795]: it is only used as "max_prot & cap_maxprot" after this point

Comment (brooksdavis) [reply sys/vm/vm_mmap.c:795]: I think @jrtc27's suggestion would be a good change.

**Review by brooksdavis — APPROVED** [review]
Fine as is, but I concur with a suggestion.

File: sys/vm/vm_mmap.c (hunk 2 above)
Comment (bukinr) [reply sys/vm/vm_mmap.c:795]: I implemented @jrtc27's suggestion on "max_prot & cap_maxprot"

**Pull request merged**